presto-main/src/main/java/com/facebook/presto/type/TimeOperators.java

```diff
@@ -122,7 +122,7 @@ public static Slice castToSlice(ConnectorSession session, @SqlType(StandardTypes
     public static long castFromSlice(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice value)
     {
         try {
-            return parseTimeWithoutTimeZone(session.getTimeZoneKey(), value.toStringUtf8());
+            return parseTime(session.getTimeZoneKey(), value.toStringUtf8());
```

Comment:
Yes. Parse time tries to parse time with timezone and if it fails it tries time without timezone.